bundles/net.tourbook/src/net/tourbook/ui/views/TourNutritionView.java

```diff
@@ -0,0 +1,1514 @@
+
+         tourId = selectionTourId.getTourId();
+
+      } else if (selection instanceof SelectionDeletedTours) {
+
+         clearView();
+      }
+
+      if (tourData == null) {
+
+         if (tourId >= TourDatabase.ENTITY_IS_NOT_SAVED) {
+
+            tourData = TourManager.getInstance().getTourData(tourId);
+            if (tourData != null) {
+               _tourData = tourData;
+            }
+         }
+      } else {
+
+         _tourData = tourData;
+      }
+
+      updateUI_ProductViewer();
+   }
+
+   private void onTableSelectionChanged() {
+
+      final StructuredSelection selection = (StructuredSelection) _productsViewer.getSelection();
+
+      _btnDeleteProduct.setEnabled(selection.size() > 0);
+   }
+
+   @Override
+   public ColumnViewer recreateViewer(final ColumnViewer columnViewer) {
+
+      _viewerContainer.setRedraw(false);
+      {
+         _productsViewer.getTable().dispose();
+
+         createUI_220_Viewer(_viewerContainer);
+         _viewerContainer.layout();
+
+         // update the viewer
+         reloadViewer();
+      }
+      _viewerContainer.setRedraw(true);
+
+      return _productsViewer;
+   }
+
+   private void refreshTourData(final TourData tourData) {
+      // get modified tour
+      _tourData = tourData;
+
+      updateUI_ProductViewer();
+
+      // removed old tour data from the selection provider
+      _postSelectionProvider.clearSelection();
+   }
+
+   @Override
+   public void reloadViewer() {
+      updateUI_ProductViewer();
+   }
+
+   private void restoreState() {
+
+      // restore old used queries
+      final String[] stateSearchedQueries = _state.getArray(STATE_PRODUCT_SEARCHES_HISTORY);
+      if (stateSearchedQueries != null) {
+         Stream.of(stateSearchedQueries).forEach(query -> _searchHistory.add(query));
+      }
+
+      _sectionSummary.setExpanded(Util.getStateBoolean(_state, STATE_SECTION_SUMMARY, true));
+      _sectionProductsList.setExpanded(Util.getStateBoolean(_state, STATE_SECTION_PRODUCTS_LIST, true));
+   }
+
+   @PersistState
+   private void saveState() {
+
+      _state.put(STATE_PRODUCT_SEARCHES_HISTORY, _searchHistory.toArray(new String[_searchHistory.size()]));
+      _state.put(STATE_SECTION_SUMMARY, _sectionSummary.isExpanded());
+      _state.put(STATE_SECTION_PRODUCTS_LIST, _sectionProductsList.isExpanded());
+   }
+
+   private void setColumnsEditingSupport() {
+
+      _colDef_ConsumedQuantity.setEditingSupport(new EditingSupport(_productsViewer) {
+
+         private SpinnerCellEditor spinnerCellEditor = new SpinnerCellEditor(_productsViewer.getTable(),
+               _nf2,
+               new RangeContent(0, 0, 0),
+               SWT.NONE);
+
+         @Override
+         protected boolean canEdit(final Object element) {
+            return true;
+         }
+
+         @Override
+         protected CellEditor getCellEditor(final Object element) {
+
+            //TODO FB it's all over the place as there are 3 possible ways
```

Comment:
@wolfgang-ch Even though I tried to configure the spinner increments, I get different behaviors depending if I use the mouse wheel, the mouse wheel on the number vs on the sign etc...
Do you know if it's a bug ? or is my code missing something ?

Reply:
I saw that but I see issues in Linux AND Windows. I will try to make a video to describe it better

Reply:
Attached is a video where I use the mouse wheel on Windows and it increments with **12** even though I specified **25** in the code !?

Reply:
Fixed but the issue with the spinner still occurs

Reply:
After debugging I discovered, that the spinner in the cell editor behaves differently compared to the normal usage. It is incrementing the value by 2 instead of 1 when the mousewheel is use.

The mousewheen do not support the value which is set  by `setIncrement` or `setPageIncrement`, this must be implemented by yourself

`Util.adjustSpinnerValueOnMouseScroll` or `UI.adjustSpinnerValueOnMouseScroll` do not work as it should in the cell editor

Reply:
> The mousewheen do not support the value which is set by `setIncrement` or `setPageIncrement`, this must be implemented by yourself
> 
> `Util.adjustSpinnerValueOnMouseScroll` or `UI.adjustSpinnerValueOnMouseScroll` do not work as it should in the cell editor

This is probably a limitation of the `SpinnerCellEditor`

Reply:
> > The mousewheen do not support the value which is set by `setIncrement` or `setPageIncrement`, this must be implemented by yourself
> > `Util.adjustSpinnerValueOnMouseScroll` or `UI.adjustSpinnerValueOnMouseScroll` do not work as it should in the cell editor
> 
> This is probably a limitation of the `SpinnerCellEditor`

It's a different behaviour.

When the selection listener method is called, then the content of the spinner is already set, in a normal selection listener the value is not yet set. This is the reason why the mousewheel is adding 2 instead of 1.

Just add a breakpoint in this listener and when it breaks, then the UI is already updated with the new value.

Reply:
So you're saying there is a way to fix that behavior ? I am not sure to understand

Reply:
In the listener, add/subtract 24 and the value is set to +/-25

Have a look into `net.tourbook.common.UI.adjustSpinnerValueOnMouseScroll()` to see how it's implemented

Reply:
Thanks, it did the trick !

Fixed